File: backend/src/models/Wishlist.js
const pool = require('../config/database');
const { v4: uuidv4 } = require('uuid');

const Wishlist = {
  async create(userId, name, description = '', isPublic = false) {
    const id = uuidv4();
    const createdAt = new Date();

    const result = await pool.query(
      'INSERT INTO wishlists (id, user_id, name, description, is_public, created_at) VALUES ($1, $2, $3, $4, $5, $6) RETURNING *',
      [id, userId, name, description, isPublic, createdAt]
    );

    return result.rows[0];
  },

  async findById(id) {
    const result = await pool.query('SELECT * FROM wishlists WHERE id = $1', [id]);
    return result.rows[0];
  },

  async findByUserId(userId) {
    const result = await pool.query('SELECT * FROM wishlists WHERE user_id = $1 ORDER BY created_at DESC', [userId]);
    return result.rows;
  },

  async update(id, name, description, isPublic) {
    const result = await pool.query(
      'UPDATE wishlists SET name = $1, description = $2, is_public = $3 WHERE id = $4 RETURNING *',
      [name, description, isPublic, id]
    );
    return result.rows[0];
  },

  async delete(id) {
    await pool.query('DELETE FROM wishlists WHERE id = $1', [id]);
  },

  async getPublic(limit = 20, offset = 0) {
    const result = await pool.query(
      'SELECT w.*, u.username FROM wishlists w JOIN users u ON w.user_id = u.id WHERE w.is_public = true ORDER BY w.created_at DESC LIMIT $1 OFFSET $2',
      [limit, offset]
    );
    return result.rows;
  },
};

module.exports = Wishlist;

File: backend/models/Wishlist.py
from config.database import Database

class Wishlist:
    """Wishlist model for database operations"""
    
    @staticmethod
    def create(user_id, name, description=None, is_public=False, follower_only=False):
        """Create a new wishlist"""
        sql = '''
            INSERT INTO wishlists (user_id, name, description, is_public, follower_only)
            VALUES (%s, %s, %s, %s, %s)
            RETURNING *
        '''
        return Database.execute_and_fetch(sql, (user_id, name, description, is_public, follower_only))
    
    @staticmethod
    def get_by_id(wishlist_id):
        """Get wishlist by ID"""
        sql = 'SELECT * FROM wishlists WHERE id = %s'
        return Database.query_one(sql, (wishlist_id,))
    
    @staticmethod
    def get_by_user(user_id):
        """Get all wishlists for a user"""
        sql = 'SELECT * FROM wishlists WHERE user_id = %s ORDER BY created_at DESC'
        return Database.query(sql, (user_id,))
    
    @staticmethod
    def get_public():
        """Get all public wishlists with pagination"""
        sql = '''
            SELECT w.*, u.username, u.avatar_url
            FROM wishlists w
            JOIN users u ON w.user_id = u.id
            WHERE w.is_public = TRUE
            ORDER BY w.created_at DESC
            LIMIT 20
        '''
        return Database.query(sql)
    
    @staticmethod
    def update(wishlist_id, **kwargs):
        """Update wishlist fields"""
        allowed_fields = ['name', 'description', 'is_public', 'follower_only']
        updates = {k: v for k, v in kwargs.items() if k in allowed_fields}
        
        if not updates:
            return None
        
        set_clause = ', '.join([f'{k} = %s' for k in updates.keys()])
        values = list(updates.values()) + [wishlist_id]
        
        sql = f'UPDATE wishlists SET {set_clause} WHERE id = %s RETURNING *'
        return Database.execute_and_fetch(sql, values)
    
    @staticmethod
    def delete(wishlist_id):
        """Delete wishlist"""
        sql = 'DELETE FROM wishlists WHERE id = %s'
        return Database.execute(sql, (wishlist_id,))
